### PR TITLE
[Volume Splitting] feat/test: group creation fee bypassing

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #6507

## What is the purpose of the change

This PR introduces a parameter that whitelists certain addresses that can bypass governance or fee for group creation.
This is a UX change for convenience of managing internal incentives.

A new `chargeGroupCreationFeeIfNotWhitelisted` method is defined. 

Parameter is initialized in the upgrade handler.

## Testing and Verifying

- `chargeGroupCreationFeeIfNotWhitelisted` unit tests added
- `CreateGroup` unit tests are updated

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/incentives/keeper/gauge_test.go [hunk 1]
```diff
@@ -44,6 +47,10 @@ var (
 		sdk.NewCoin("uosmo", osmomath.NewInt(100000)),
 		sdk.NewCoin("atom", osmomath.NewInt(99999)),
 	)
+
+	customGroupCreationFee = sdk.NewCoins(sdk.NewInt64Coin(feeDenom, 1000000))
+
+	errorNoCustomFeeInBalance = fmt.Errorf("0%s is smaller than %s: insufficient funds", feeDenom, customGroupCreationFee)
```
Comment (AlpinYukseloglu) [inline x/incentives/keeper/gauge_test.go:53]: Out of curiosity, why don't we have a custom error type here? Is it just to avoid PR scope creep with driveby changes?

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM, nice job! Since this is a quite similar structure to admin pool creation, it might be worth starting to think about a good way to abstract this logic and make it easier to implement new subdao admin functionality without needing to redo all this param boilerplate work

File: x/incentives/keeper/gauge_test.go (hunk 1 above)
Comment (p0mvn) [reply x/incentives/keeper/gauge_test.go:53]: This is because the error originates in the bank module which does not export it.

As a result, I'm forced to redefine here and do error string comparison as opposed to proper error comparison

**Comment by czarcas7ic** [comment]
@AlpinYukseloglu yeah we discussed that we should be looking into potentially creating a module to allow whitelisted actions for DAO addresses across all modules, I think it would be really useful

File: x/incentives/keeper/gauge.go [hunk 2]
```diff
@@ -298,6 +294,35 @@ func (k Keeper) CreateGroup(ctx sdk.Context, coins sdk.Coins, numEpochPaidOver u
...
+		unrestrictedAddress, err := sdk.AccAddressFromBech32(unrestrictedAddressStr)
+		if err != nil {
+			return false, err
+		}
+
+		// sender is either unrestrictedAddress or the module account
+		if unrestrictedAddress.Equals(sender) || sender.Equals(incentivesModuleAddress) {
+			return false, nil
+		}
+	}
```
Comment (czarcas7ic) [inline x/incentives/keeper/gauge.go:316]: wdyt about adding the module address to this whitelist param instead of retrieving it on every call? Can ignore, would be minor perf.

Comment (p0mvn) [reply x/incentives/keeper/gauge.go:316]: I prefer the explicitness of getting a module account for readability. If it's in params, it's not obvious and might make it a hard-to-debug detail in the future

File: x/incentives/keeper/msg_server_test.go [hunk 3]
```diff
@@ -109,13 +110,19 @@ func (s *KeeperTestSuite) TestCreateGauge_Fee() {
 			Duration:      defaultLockDuration,
 		}
 
+		// TODO: consider removing isPerpetual param
+		numEpochsPaidOver := incentiveskeeper.PerpetualNumEpochsPaidOver
+		if !tc.isPerpetual {
+			numEpochsPaidOver++
+		}
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply x/incentives/keeper/msg_server_test.go:117]: I thought I removed this, I will come back and remove, thanks for catching

[the remark this answers is not in the document]
Comment (p0mvn) [reply x/incentives/keeper/msg_server_test.go]: done
